safenode/src/network/event.rs

```diff
@@ -107,7 +110,7 @@ impl SwarmDriver {
                     let new_peers: HashSet<PeerId> =
                         closest_peers.peers.clone().into_iter().collect();
                     current_closest.extend(new_peers);
```

Comment:
Is this check still necessary when we set replication factor for the kad cfg?
I don't know the connection there. How do those things relate (i.e. tthe kad cfg replication factor, and what is going on in this code block).

Reply:
I have my doubts about this event as a whole tbh. When we use the `SwarmCmd::GetClosestPeers` to ask the swarm for the peers, the function reads that  "The result of the query is delivered in  `KademliaEvent::OutboundQueryCompleted{QueryResult::GetClosestPeers}` ". But there is no such `OutboundQueryCompleted`  variant inside `KademliaEvent`, instead we just have the `OutboundQueryProgressed` event?

So which is the truth here? Do the `Progressed` event actually means that we get the final result after querying the whole network? I don't see the `OutboundQueryProgressed` repeated as well, ie the network send one and only event back.

Reply:
Digging futher, the `KademliaEvent::OutboundQueryProgressed` is emitted by a function called `Kademlia::query_finished`.

Reply:
There are flags within the returned result to indicate the status of the query and we shall define the `completion criterias` based on those.
There are TODO poins below mention that?

even with the changing of the kad cfg, this may still needed with a larger network for a more accurate result.